app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -469,7 +469,13 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
                 ImageCaptureProcessing.processCaptureResponse(this, getInstanceFolder(), true);
                 break;
             case SIGNATURE_CAPTURE:
-                ImageCaptureProcessing.processCaptureResponse(this, getInstanceFolder(), false);
+                boolean saved = ImageCaptureProcessing.processCaptureResponse(this, getInstanceFolder(), false);
+                if (saved) {
+                    ImageButton nextButton = (ImageButton)this.findViewById(R.id.nav_btn_next);
+                    if (nextButton.getTag().equals(NAV_STATE_NEXT) && !questionsView.isQuestionList()) {
```

Comment:
`nextButton.getTag().equals(NAV_STATE_NEXT) ` is pretty 'low-level' code, it would be good to lift it into a named helper method and also use it in `onForwardSwipe`, which has the same line of code.